### PR TITLE
New three.js version

### PR DESCRIPTION
**Pull request by schnellecom**
As discussed today with Prof. Niemeyer, here is the PR to have the newly implemented version of three.js in a branch on the main repository.

### CODE REVIEW

File: gap/PolygonalComplexes/animating.gd [hunk 1]
```diff
@@ -52,7 +52,27 @@
...
+#! The files output by <K>DrawSurfaceToJavaScript</K> have some options that can be changed live in the html file. 
+#! These are in the graphical user interface (GUI). Here is an example:
+
+#! <Alt Only="JavaScript">
+#! GUI.html
+#! </Alt>
+
+#! Here you can change the following:
+#! <Enum>
+#!   <Item>The speed of rotation in the X, Y and Y direction</Item>
```
Comment (MeikeWeiss) [inline gap/PolygonalComplexes/animating.gd:69]: X, Y and Z direction

File: gap/PolygonalComplexes/animating.gi [hunk 2]
```diff
@@ -797,170 +806,302 @@ InstallMethod( GetCircleColours,
...
+        AppendTo(output, """
+            const mesh""",i,""" = new THREE.Mesh( geometry""",i,""", material""",i,""" );
+            mesh""",i,""".castShadow = true;                         
+            mesh""",i,""".receiveShadow = true;                      
+                                        
+            meshRoot.add(mesh""",i,""");
+        """);
+    od;
+
+    # generate a wireframe from the faces
```
Comment (MeikeWeiss) [inline gap/PolygonalComplexes/animating.gi:949]: Can this commented out part be removed?

File: gap/PolygonalComplexes/animating.gi [hunk 3]
```diff
@@ -797,170 +806,302 @@ InstallMethod( GetCircleColours,
...
+                color: """,color,""",
+                linewidth: """,edgeThickness,""",
+            } );        
+        """);
+
+        AppendTo(output, "\tconst edges",i," = new Float32Array( [");
+
+        colorPositions := Positions(edgeColors, color);
+        for edge in colorPositions do
+            # if IsEdgeActive(surface, edge, printRecord) then
```
Comment (MeikeWeiss) [inline gap/PolygonalComplexes/animating.gi:992]: Can this commented out line be removed?

File: gap/PolygonalComplexes/animating.gi [hunk 4]
```diff
@@ -797,170 +806,302 @@ InstallMethod( GetCircleColours,
...
+        AppendTo(output, """
+            const edgeGeometry""",i,""" = new THREE.BufferGeometry();
+            edgeGeometry""",i,""".setAttribute( 'position', new THREE.BufferAttribute( edges""",i,""", 3 ) );
+
+            const edgeLine""",i,""" = new THREE.LineSegments( edgeGeometry""",i,""", edgeMaterial""",i,""" );
+            edgeRoot.add(edgeLine""",i,""");
+        """);
+        
+    od;
+    # for i in [1..Length(faces)] do
```
Comment (MeikeWeiss) [inline gap/PolygonalComplexes/animating.gi:1026]: Can this commented out part be removed?

**Review by MeikeWeiss — COMMENTED** [review]
Nice!

File: gap/PolygonalComplexes/animating.gi (hunk 2 above)
Comment (schnellecom) [reply gap/PolygonalComplexes/animating.gi]: Removed in b3afa4d

File: gap/PolygonalComplexes/animating.gi (hunk 3 above)
Comment (schnellecom) [reply gap/PolygonalComplexes/animating.gi:982]: I would prefer to keep it in, as once schnellecom/SimplicialSurfaces#18 is resolved the check should be reenabled.

File: gap/PolygonalComplexes/animating.gi (hunk 4 above)
Comment (schnellecom) [reply gap/PolygonalComplexes/animating.gi]: Removed in [e5ec4ce](https://github.com/gap-packages/SimplicialSurfaces/pull/248/commits/e5ec4ce7bd21dde09732f1ffb3b2b66b7d9f3c2f)

**Review by schnellecom — COMMENTED** [review]
Here I got confused for a second, thinking the ```end``` would belong to the merge. Fixed in a863975

File: gap/PolygonalComplexes/animating.gd [hunk 5]
```diff
@@ -52,7 +52,27 @@
...
+#! The file can be opened with most browsers and then the surface is shown as a 3D render with some manipulation options. 
+#! Because we load the three.js dependecies (and three.js itself) from a CDN (content delivery network) it requires an active internet connection.
+
+#! @Section Output files
+#! @SectionLabel output
+#! The files output by <K>DrawSurfaceToJavaScript</K> have some options that can be changed live in the html file. 
+#! These are in the graphical user interface (GUI). Here is an example:
+
+#! <Alt Only="JavaScript">
+#! GUI.html
```
Comment (MeikeWeiss) [inline gap/PolygonalComplexes/animating.gd:64]: This picture isn't shown in the manual

**Pull request merged**